src/viem/saga.ts

```diff
@@ -135,7 +199,7 @@ export function* sendPayment({
  */
 function* getTransferSimulateContract({
   wallet,
-  tokenAddress,
+  tokenId,
```

Comment:
why refactor this to take token Id's if it should never be called for native assets? should it take token address and network id instead?

Reply:
seems like requiring those two fields would give us static validation instead of runtime errors, which would be preferable

Reply:
Yeah, after going back and forth on this, I've updated this to take in a `tokenAddress` and a `tokenInfo` object, which gives us everything we need, and means we don't have to use the deprecated `byAddress` hooks to get the full token info object.